sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -682,6 +686,12 @@ export class StreamingReceiver extends MessageReceiver {
     }
   }
 
+  async close(): Promise<void> {
+    await super.close();
+    this.isReceivingMessages = false;
+    this.stopReceivingMessages();
```

Comment:
Do we allow re-opening a streamingReceiver instance after `close()` is called on it? The way I see it the answer is no. If we make `isReceiveingMessages` a getter so that it returns value from the helper which in turn can check if receiver is open or not, then we do not need the above two lines and can avoid overriding the close() method right?

Reply:
Also, I wonder if this close() should do a `this.context.streamingReceiver = undefined`

The existence of `context.streamingReceiver` makes our connection recovery logic in `ClientEntityContext` to call `onDetached()` on it to recover the streaming receiver. Why would we want to recover a closed Streaming Receiver?

Reply:
These are all good points. Taking a look.

Reply:
First point (about just allowing isReceivingMessages) is easy enough - that's been pushed.

The second one - I'd want to spend a bit more time looking into the implications of unsetting context.streamingReceiver - logically it makes sense but I'd like to approach that in a separate PR.

Reply:
Second one filed as issue #10481